mobileapps/tests.py

```diff
@@ -0,0 +1,254 @@
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['count'], 0)
+
+    def test_mobileapps_list_search_by_organization_name(self):
+        mobile_apps = []
+        organization1 = (Organization.objects.create(name='ABC Organization'))
+        organization2 = (Organization.objects.create(name='XYZ Organization'))
+
+        mobile_apps.append(self.setup_test_mobileapp(
+            mobileapp_data={
+                "name": 'ABC App', "identifier": "ABC", "organizations": [organization1.id, organization2.id]})
+        )
+
+        mobile_apps.append(self.setup_test_mobileapp(
+            mobileapp_data={
+                "name": 'XYZ App', "identifier": "XYZ", "organizations": [organization2.id]})
+        )
+
+        test_uri = '{}'.format(self.base_mobileapps_uri)
+        response = self.do_get('{}?organization_name=xyz'.format(test_uri))
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['count'], 2)
+        self.assertEqual(response.data['results'][0]['identifier'], "ABC")
+        self.assertEqual(response.data['results'][1]['identifier'], "XYZ")
+
+        response = self.do_get('{}?organization_name=abc'.format(test_uri))
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['count'], 1)
+        self.assertEqual(response.data['results'][0]['identifier'], "ABC")
+
+        response = self.do_get('{}?organization_name=query'.format(test_uri))
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['count'], 0)
+
+    def test_mobileapps_detail_get(self):
+        mobileapp_data = self.setup_test_mobileapp(mobileapp_data={"name": 'ABC App', "identifier": 'ABC'})
+
+        response = self.do_get(reverse('mobileapps-detail', kwargs={'pk': mobileapp_data['id']}))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['name'], 'ABC App')
+        self.assertEqual(response.data['identifier'], 'ABC')
+        self.assertEqual(response.data['updated_by'], self.test_mobileapp_updated_by)
+        self.assertEqual(response.data['is_active'], True)
+        self.assertIn("operating_system", response.data)
+        self.assertIn("deployment_mechanism", response.data)
+        self.assertIn("download_url", response.data)
+        self.assertIn("analytics_url_dev", response.data)
+        self.assertIn("analytics_url_prod", response.data)
+        self.assertIn("notification_provider", response.data)
+        self.assertIn("provider_key", response.data)
+        self.assertIn("provider_secret", response.data)
+        self.assertIn("provider_dashboard_url", response.data)
+        self.assertIn("current_version", response.data)
+        self.assertIsNotNone(response.data['created'])
+        self.assertIsNotNone(response.data['modified'])
+
+    def test_mobileapps_detail_put(self):
+
+        mobileapp = MobileApp.objects.create(
```

Comment:
why not using  `self.setup_test_mobileapp(mobileapp_data=data)` method to create mobiel app?

Reply:
I am using this just to ease myself by getting the object not response. It is easier to get values from object which I have to use after creation.